docs/source/key_components/rl_toolkit.rst

```diff
@@ -58,138 +54,115 @@ easily turned into a roll-out worker or an actor for synchronous and asynchronou
 
-Moreover, in ``data-parallel`` mode, there will be an additional ``task_queue`` to perform auto-balance.
-For each training iteration, policies first send requests to ``task_queue``, then ``task_queue`` assigns
-workers according to the idle status. After assigned a set of ``grad_worker`` IDs, policies submit training
-tasks to their own workers. The ``task_queue`` actually processes task requests as a queue.
+Users should choose the proper types of models according to the type of policies, and then implement their own
+models by inherit the abstract ones (just like ``MyQNet``).
 
-.. image:: ../images/rl/policy_manager.svg
-   :target: ../images/rl/policy_manager.svg
-   :alt: PolicyManager
+There are also some other models for training purpose. For example:
 
-The ``DistributedPolicyManager`` runs a set of ``policy_host``.
-``policy_host`` is a process/VM/node that hosts the update of a policy.
-Each ``policy_host`` independently updates its own policies for policy-level parallelism.
+- ``VNet``: Used in the critic part in the actor-critic algorithm.
+- ``MultiQNet``: Used in the critic part in the MADDPG algorithm.
+- ...
 
-During training, the ``PolicyManager`` receives training data collected by the ``RolloutManager``,
-then send them to corresponding ``policy_host``. Each ``policy_host`` will send gradient tasks consist
-of policy state and experience batch, to several stateless ``grad_worker`` for gradient computation.
-The ``grad_worker`` is stateless, and computes gradients using the policy state and data
-batch provided in a task.
-Then ``policy_host`` aggregates the gradients from ``grad_worker`` s, and performs gradient descent
-on its parameters.
+The way to use these models is exactly the same as the way to use the policy models.
 
-Core Model
-----------
+.. _trainer:
 
-In the deep reinforcement learning (DRL) world, a policy usually includes one or more neural-network com-based models,
-which may be used to compute action preferences or estimate state / action values. The ``AbsCoreModel`` represents a
-collection of network components with embedded optimizers and exposes unified interfaces to decouple model inference
-and optimization from the algorithmic aspects of the policy that uses them. For example, the actor-critic algorithm
-does not need to concern itself with how the action probabilities and state values are computed. Subclasses of
-``AbsCoreModel`` provided for use with specific RL algorithms include ``DiscreteQNet`` for DQN, ``DiscretePolicyNet``
-for Policy Gradient, ``DiscreteACNet`` for Actor-Critic and ``ContinuousACNet`` for DDPG.
+Trainer
+-------
 
-The code snippet below shows how to create a model for the actor-critic algorithm with a shared bottom stack:
+When introducing policies, we mentioned that policies cannot train themselves. Instead, they have to be trained
+by external trainers. In MARO, a trainer is corresponding to a kind of RL algorithm, such as DQN, actor-critic,
```

Comment:
_a trainer corresponds to an RL algorithm_ or _a trainer represents an RL algorithm_?